File: 백준/Silver/2839. 설탕 배달/설탕 배달.py
import sys

# input = sys.stdin.readline

n = int(input())
#3,5 두개 있음
cnt = 0
while n > 0:
    if n % 5 == 0:
        cnt += n // 5
        n = n%5
        break
    else:
        n -= 3 
        cnt += 1
if n == 0:
    print(cnt)
else:
    print(-1)

File: 백준/Silver/2839. 설탕 배달/설탕 배달.java
import java.io.BufferedReader;
import java.io.FileInputStream;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class Main {

    // ! 3,5봉지로 정확하게 n 얻기 + 최대한 적은 봉지 사용
    static int n;

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st;

        n = Integer.parseInt(br.readLine());
        int cnt = 0;
        boolean flag = true;
        while (n > 0) {
            if (n % 5 == 0) {
                cnt += n / 5;
                n /= 5;
                break;
            } else {
                cnt += 1;
                n -= 3;
            }
        }
        if(n < 0) System.out.println(-1);
        else System.out.println(cnt);
    }

}